.circleci/config.yml

```diff
@@ -179,20 +180,76 @@ jobs:
           paths:
             - ~/yt_test
 
-      - lint
-
       - restore_cache:
           name: "Restore test answers."
           key: python-<< parameters.tag >>-test-answers-2021-22-10
 
-      - build-and-test
+      - build-and-test-nose
 
       - save_cache:
           name: "Save test answers cache."
           key: python-<< parameters.tag >>-test-answers-2021-22-10
           paths:
             - ~/test_results
 
+  run-tests-pytest:
+    # Run a subset of the test suite (yield-based tests are not supported by pytest)
+    # this is necessary to test in Python 3.10 because it's not compatible with nose
+    parameters:
+      tag:
+        type: string
+        default: latest
+    executor:
+      name: python
+      tag: << parameters.tag >>
+
+    working_directory: ~/yt_astro_analysis
+
+    steps:
+      - checkout
+      - set-env
+
+      - restore_cache:
+          name: "Restore dependencies cache."
+          key: python-<< parameters.tag >>-dependencies-2021-22-10
+
+      - install-with-yt-dev
+
+      - save_cache:
+          name: "Save dependencies cache"
+          key: python-<< parameters.tag >>-dependencies-2021-22-10
+          paths:
+            - ~/.cache/pip
+            - ~/venv
+            - ~/yt-git
+            - ~/rockstar-galaxies
+
+      - restore_cache:
+          name: "Restore test data cache."
+          key: test-data-2021-23-03
+
+      - download-test-data
+
+      - save_cache:
+          name: "Save test data cache."
+          key: test-data-2021-23-03
+          paths:
+            - ~/yt_test
+
+      - restore_cache:
+          name: "Restore test answers."
+          key: python-<< parameters.tag >>-test-answers-2021-22-10
+
+      - build-and-test-pytest
+
+      # Commenting this out before review because I'm not sure how well this can
```

Comment:
Perhaps we can keep this in by changing the key name to something like:
```
pytest-<< parameters.tag >>-test-answers-2021-22-10
```
and for nose:
```
nose-<< parameters.tag >>-test-answers-2021-22-10
```

Reply:
sounds good, thanks